src/gc.c

```diff
@@ -3029,6 +3029,9 @@ void jl_gc_init(void)
     arraylist_new(&finalizer_list_marked, 0);
     arraylist_new(&to_finalize, 0);
 
+    assert(jl_n_threads > 0);
+    default_collect_interval = default_collect_interval * jl_n_threads;
```

Comment:
Is linear scaling advisable? Now we have much longer interval in serial sections of the code, leading to generally more memory usage. Maybe some form of sublinear scaling would make sense.

Reply:
Just depends how much time you'd like to spend in GC :) So far I've found that anything less than this means more threads => greater % GC time and it's quite depressing when you're trying to get something to scale.